src/editor/rotation_bounds_handle.cpp

```
@@ -0,0 +1,140 @@
+    auto const dx = static_cast<float>(pos.x) - drawPos.x;
+    auto const dy = static_cast<float>(pos.y) - drawPos.y;
+    auto const halfSize = m_size / 2.0f;
+    return (dx * dx + dy * dy) <= (halfSize * halfSize);
+}
+
+void RotationBoundsHandle::UpdatePosition(moth_ui::IntVec2 const& position) {
+    auto const pivotWorld = GetPivotWorldPos();
+    auto const mousePos = static_cast<moth_ui::FloatVec2>(position);
+    auto const dx = mousePos.x - pivotWorld.x;
+    auto const dy = mousePos.y - pivotWorld.y;
+    float const currentAngle = std::atan2(dy, dx);
```

Clang (14.0.6)

[error] variable 'currentAngle' is not initialized

(cppcoreguidelines-init-variables,-warnings-as-errors)